src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -141927,5 +141927,376 @@ static T GetValue2<T>(Collection c, object key)
+        {
+            var src = @"
+#nullable disable
+interface I<T> { T Item { get; } }
+
+#nullable enable
+interface I2<T> : I<T> { }
+
+#nullable disable
+interface I3 : I<(object a, object b)>, I2<(object notA, object notB)> { }
+
+#nullable enable
+public class C
+{
+    void M(I3 i)
+    {
+        _ = i.Item;
+    }
+}
+";
+            var comp = CreateCompilation(src);
+            comp.VerifyDiagnostics(
+                // (9,11): error CS8140: 'I<(object notA, object notB)>' is already listed in the interface list on type 'I3' with different tuple element names, as 'I<(object a, object b)>'.
+                // interface I3 : I<(object a, object b)>, I2<(object notA, object notB)> { }
+                Diagnostic(ErrorCode.ERR_DuplicateInterfaceWithTupleNamesInBaseList, "I3").WithArguments("I<(object notA, object notB)>", "I<(object a, object b)>", "I3").WithLocation(9, 11),
+                // (16,15): error CS0229: Ambiguity between 'I<(object a, object b)>.Item' and 'I<(object notA, object notB)>.Item'
+                //         _ = i.Item;
+                Diagnostic(ErrorCode.ERR_AmbigMember, "Item").WithArguments("I<(object a, object b)>.Item", "I<(object notA, object notB)>.Item").WithLocation(16, 15)
+                );
+            var i3 = comp.GetTypeByMetadataName("I3");
+            AssertEx.Equal(new string[] { "I<(object a, object b)>", "I2<(object notA, object notB)>" },
+                i3.Interfaces().ToTestDisplayStrings(TypeWithAnnotations.TestDisplayFormat));
+            AssertEx.Equal(new string[] { "I<(object a, object b)>", "I2<(object notA, object notB)>", "I<(object notA, object notB)>" },
+                i3.AllInterfaces().ToTestDisplayStrings(TypeWithAnnotations.TestDisplayFormat));
+        }
+
+        [Fact]
+        public void AmbigMember_NoDifference()
+        {
+            var src = @"
+interface I<T> { T Item { get; } }
+
+interface I2<T> : I<T> { }
+
+interface I3 : I<object>, I2<object> { }
+
+public class C
+{
+    void M(I3 i)
+    {
+        _ = i.Item;
+    }
+}
+";
+            var comp = CreateCompilation(src);
+            comp.VerifyDiagnostics(
+                );
+            var i3 = comp.GetTypeByMetadataName("I3");
+            AssertEx.Equal(new string[] { "I<object>", "I2<object>" },
+                i3.Interfaces().ToTestDisplayStrings(TypeWithAnnotations.TestDisplayFormat));
+            AssertEx.Equal(new string[] { "I2<object>", "I<object>" },
+                i3.AllInterfaces().ToTestDisplayStrings(TypeWithAnnotations.TestDisplayFormat));
+        }
+
+        [Fact]
+        public void AmbigMember_DifferenceBetweenNonnullableAndOblivious_WithoutConstraint()
+        {
+            var src = @"
+#nullable disable
+interface I<T> { T Item { get; set; } }
+
+#nullable enable
+interface I2<T> : I<T> { }
+
+#nullable disable
+interface I3 : I<object>, I2<object> { }
+
+#nullable enable
+public class C
+{
+    void M(I3 i)
+    {
+        _ = i.Item;
+        i.Item = null;
```

Comment:
>Item [](start = 10, length = 4)

Consider expanding testing to other kinds of members: fields, indexers, events, nested types, methods.